src/extension.ts

```diff
@@ -37,7 +38,7 @@ export async function activate(context: vscode.ExtensionContext): Promise<IExten
         promptForConnection: (ignoreFocusOut?: boolean) => {
             return controller.connectionManager.connectionUI.promptForConnection(ignoreFocusOut);
         },
-        connect: async (connectionInfo: IConnectionInfo) => {
+        connect: async (connectionInfo: IConnectionInfo, saveConnection?: boolean) => {
```

Comment:
IConnectionProfile already has saveConnection on it...so I think instead you should be looking at utilizing that. Maybe by moving that interface into the API type declarations...although it's kind of weird to have these two types then that are basically the same thing.

Reply:
I was basically trying to be consistent with ADS. ADS has connect API with saveConnection in the method. Although it's also part of connection profile

Reply:
well that's what being consistent means :) if we decided that's how to same a connection in ADS then I assume we thought it over. but if that's not the right way to do then I can also add a separate API for saving connection and get the IConnectionProfile as input, this way I won't have problem with casting. Also I need to add an option to save in workspace or user setting, it would be easier to add that into a new API function

Reply:
I'm guessing in future when we work with projects in a workspace, we might want to store the connections for the project in the workspace so it can be shared with others using the workspace. but I'm now doing anything on that right now since nothing is final